app/renderer/js/utils/logger-util.js

```diff
@@ -30,6 +30,20 @@ class Logger {
 			timestamp = this.getTimestamp;
 		}
 
+		fs.readFile(file, 'utf8', (err, data) => {
+			if (err) {
+				throw err;
+			}
+			const logs = data.split('\n');
+			// Keep top 300 files of log for each instance
+			const trimLines = logs.slice(Math.max(logs.length - 300, 0));
+			let toWrite = '';
+			trimLines.forEach(line => {
+				toWrite += line + '\n';
+			});
```

Comment:
You don't need to use a forEach loop here just use `trimmedLogs.join(os.EOL)` here to get the log file in string format.

Reply:
Yeah, I think that should remove the computational complexity.